### PR TITLE
refactor: reuse timeout channel creation

### PR DESCRIPTION
**Pull request by suzaku**

### CODE REVIEW

**Review by penberg — APPROVED** [review]
lgtm!

**Review by fzyzcjy — APPROVED** [review]

**Comment by fzyzcjy** [comment]
oops please ignore my approval (mistakenly click that)

File: little_raft/src/timer.rs [hunk 1]
```diff
@@ -9,30 +9,27 @@ pub struct Timer {
...
-        self.rx = rx;
+        self.rx = get_timeout_channel(self.timeout);
     }
 
     pub fn get_rx(&self) -> &Receiver<()> {
         &self.rx
     }
 }
+
+fn get_timeout_channel(timeout: Duration) -> Receiver<()> {
```
Comment (andreev-io) [inline little_raft/src/timer.rs:27]: Let's move this function into `impl Timer` and just keep it private and call it with `Timer::get_timeout_channel(self.timeout)`.

**Review by andreev-io — CHANGES_REQUESTED** [review]

**Review by andreev-io — APPROVED** [review]

**Comment by andreev-io** [comment]
Congrats on your first contribution to Little Raft, @suzaku!

**Comment by suzaku** [comment]
> Congrats on your first contribution to Little Raft, @suzaku!

Let's hope it's not the only one. I code in Go and Python mostly, but I'm practicing with Rust lately, if you have some easy to pick up issues, please let me know.

**Pull request merged**